kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/validation/impl/AssetRule.java

```diff
@@ -146,4 +146,47 @@ protected boolean processCustomSaveDocumentBusinessRules(MaintenanceDocument doc
 		
 		return valid;
 	}
+	
+	@Override
+    protected boolean processAssetValidation(MaintenanceDocument document) {
+        boolean valid = true;
+
+        // validate Inventory Status Code.
+        if (!StringUtils.equalsIgnoreCase(oldAsset.getInventoryStatusCode(), newAsset.getInventoryStatusCode())) {
+            valid &= validateInventoryStatusCode(oldAsset.getInventoryStatusCode(), newAsset.getInventoryStatusCode());
+        }
+
+        // validate Organization Owner Account Number
+        if (!StringUtils.equalsIgnoreCase(oldAsset.getOrganizationOwnerAccountNumber(), newAsset.getOrganizationOwnerAccountNumber())) {
+            valid &= validateAccount();
+        }
+
+        // validate asset representative name
+        if (ObjectUtils.isNotNull(oldAsset.getAssetRepresentative())
+                && ObjectUtils.isNotNull(newAsset.getAssetRepresentative())
+                && ObjectUtils.isNotNull(newAsset.getAssetRepresentative().getPrincipalName())
```

Comment:
Two thoughts here:
1. An improvement, replace: 
`ObjectUtils.isNotNull(newAsset.getAssetRepresentative().getPrincipalName())` 
with:
`StringUtils..isNotBlank(newAsset.getAssetRepresentative().getPrincipalName())`
Rationale: getPrincipalName() is a String, and moreover, a String that is not in the DB (it's pulled from LDAP)

2. You check for this on the newAsset, but not the old asset; you could fail out faster if you would. Said another way, I'd cover your oldAsset like you do on your newAsset, which would mean one more clause in your chained conditions.